Updated docs for ignore_other_prefixes to clarify behavior (#5592)
apollographql/router: docs/source/configuration/authn-jwt.mdx

</td>
<td>

> By default, the Router errors when an unknown prefix is provided in the `Authorization` header. You must explicitly define prefixes in `header_value_prefix` or `sources`.

Reviewer: I think it's helpful to start with why you might want to use this option:

```suggestion
This option lets you have a mix of `Authorization` header schemes, for example, both `Basic` and `Bearer`, without requiring you to use an another header.

By default, the router responds with an error when it encounters an unknown prefix in the `Authorization` header. You must explicitly define prefixes in [`header_value_prefix`](#header_value_prefix) or [`sources`](#sources).
```

> If you set `ignore_other_prefixes` to `true`, the Router allows unknown prefixes in the `Authorization` header and does not respond with an error when encountering one.
> 
> If you set `ignore_other_prefixes` to `false` or leave it unspecified, the Router assumes the default behavior of responding with an error when it finds an unknown prefix in the `Authorization` header.

Reviewer: I think starting with the default (and being able to refer to the default behavior just described) makes this a little easier to understand.

```suggestion
When `ignore_other_prefixes` is `false` (the default value), the router uses the default behavior and errors when it encounters an unknown prefix in the `Authorization` header.

If you set `ignore_other_prefixes` to `true`, the router allows requests with unknown prefixes in the `Authorization` header through and does not respond with an error when encountering one.
```

> If you set a header prefix to an empty string, the Router ignores this option.

Reviewer: "this option" can be a bit vague here:

```suggestion
If you set `header_value_prefix` to an empty string, the router ignores the `ignore_other_prefixes` setting.
```
